### PR TITLE
App target frameworks fix

### PR DESCRIPTION
**Pull request by DianaSoltani**
## Description
Addresses some MSBuild and NuGet TFM bugs. We want to properly refer to .android.csproj and .ios.csproj with the appropriate target framework formatting. 
The mono and ios frameworks have been updated to net7.
This improves on Nuget package compatibility checks for the Android and iOS platforms.

Also updated the package mapper to address [#1434](https://github.com/dotnet/upgrade-assistant/issues/1434)

### CODE REVIEW

File: src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/VisualStudioFinder.cs [hunk 1]
```diff
@@ -35,35 +35,32 @@ public void Configure(WorkspaceOptions options)
 
         private (string? Path, int? Version) GetLatestVisualStudioPath(string? suppliedPath)
         {
-            var latest = GetLatestPath(suppliedPath);
+            var latest = GetLatestVisualStudio(suppliedPath);
```
Comment (mgoertz-msft) [inline src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/VisualStudioFinder.cs:39]: Changes in this file seem unrelated

File: src/extensions/try-convert/MSBuild.Abstractions/ProjectExtensions.cs [hunk 2]
```diff
@@ -50,8 +50,8 @@ public static string GetTargetFramework(this IProject project)
                 ".NETCore" => "net", // UWP
                 ".NETCoreApp" => "netcoreapp",
                 ".NETPortable" => "netstandard",
-                "MonoAndroid" => "net",
-                "Xamarin.iOS" => "net",
+                "MonoAndroid" => "net7.0-android",
+                "Xamarin.iOS" => "net7.0-ios",
```
Comment (jstedfast) [inline src/extensions/try-convert/MSBuild.Abstractions/ProjectExtensions.cs:54]: I suspect that we don't want to force net7.0 here. Is there a way to get what version of .net core that the user wants to update to? e.g. we probably want to use "6.0", "7.0" or "8.0" (now that net8.0 is out) depending on what the user wants to target.

That said, on line 66 someone has already assumed "net6.0-windows" for Windows.

**Review by jstedfast — APPROVED** [review]

File: src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/PackageMaps/XamarinPackageMap.json
Comment (mgoertz-msft) [inline src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/PackageMaps/XamarinPackageMap.json:1]: Let's use the latest [2.88.3](https://www.nuget.org/packages/SkiaSharp.Views.Maui.Controls/2.88.3) for the SkiaSharp ones too

File: src/extensions/try-convert/MSBuild.Abstractions/ProjectExtensions.cs (hunk 2 above)
Comment (mgoertz-msft) [reply src/extensions/try-convert/MSBuild.Abstractions/ProjectExtensions.cs:54]: Everything for MAUI is hardcoded to 7.0 in the CLI, which is fine until 8.0 ships.

**Review by mgoertz-msft — APPROVED** [review]

**Review by abpiskunov — APPROVED** [review]
